packages/wonder-blocks-layout/components/media-layout.test.js

```diff
@@ -0,0 +1,403 @@
+import React from "react";
+import {StyleSheet} from "aphrodite";
+import {View} from "@khanacademy/wonder-blocks-core";
+
+import {mount, unmountAll} from "../../../utils/testing/mount.js";
+import MediaLayout from "./media-layout.js";
+import {resizeWindow, matchMedia} from "../util/test-util.js";
+
+describe("MediaLayout", () => {
+    beforeEach(() => {
+        unmountAll();
+        window.matchMedia = matchMedia;
+    });
+
+    describe("mediaSize", () => {
+        it("should be small when viewport width < 768", async () => {
+            // Arrange
+            resizeWindow("small");
+
+            // Act
+            const args = await new Promise((resolve, reject) => {
+                mount(
+                    <MediaLayout>
+                        {({mediaSize, mediaSpec, styles}) => {
+                            resolve({mediaSize, mediaSpec, styles});
+                            return (
+                                <View style={styles.test}>Hello, world!</View>
+                            );
+                        }}
+                    </MediaLayout>,
+                );
+            });
+
+            // Assert
+            expect(args.mediaSize).toEqual("small");
+        });
+
+        it("should be medium when viewport 768 < width < 1024", async () => {
+            // Arrange
+            resizeWindow("medium");
+
+            // Act
+            const args = await new Promise((resolve, reject) => {
+                mount(
+                    <MediaLayout>
+                        {({mediaSize, mediaSpec, styles}) => {
+                            resolve({mediaSize, mediaSpec, styles});
+                            return (
+                                <View style={styles.test}>Hello, world!</View>
+                            );
+                        }}
+                    </MediaLayout>,
+                );
+            });
+
+            // Assert
+            expect(args.mediaSize).toEqual("medium");
+        });
+
+        it("should be medium when viewport width > 1024", async () => {
+            // Arrange
+            resizeWindow("large");
+
+            // Act
+            const args = await new Promise((resolve, reject) => {
+                mount(
+                    <MediaLayout>
+                        {({mediaSize, mediaSpec, styles}) => {
+                            resolve({mediaSize, mediaSpec, styles});
+                            return (
+                                <View style={styles.test}>Hello, world!</View>
+                            );
+                        }}
+                    </MediaLayout>,
+                );
+            });
+
+            // Assert
+            expect(args.mediaSize).toEqual("large");
+        });
+    });
+
+    describe("styleSheets", () => {
+        const testSizes = {
+            small: [640, 480],
+            medium: [800, 600],
+            large: [1200, 800],
+        };
+
+        for (const size of Object.keys(testSizes)) {
```

Comment:
Not convinced this loops is worth it. Not enough to hold up the review, but I'd prefer distinct test cases than clever test generation tricks.

Reply:
I'm going to leave these for now.